pkg/platform/kube/controller/evictedpodsmonitoring.go

```diff
@@ -0,0 +1,134 @@
+/*
+Copyright 2017 The Nuclio Authors.
+
+Licensed under the Apache License, Version 2.0 (the "License");
+you may not use this file except in compliance with the License.
+You may obtain a copy of the License at
+
+    http://www.apache.org/licenses/LICENSE-2.0
+
+Unless required by applicable law or agreed to in writing, software
+distributed under the License is distributed on an "AS IS" BASIS,
+WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+See the License for the specific language governing permissions and
+limitations under the License.
+*/
+
+package controller
+
+import (
+	"context"
+	"strings"
+	"time"
+
+	"github.com/nuclio/nuclio/pkg/common"
+
+	"github.com/nuclio/logger"
+	apierrors "k8s.io/apimachinery/pkg/api/errors"
+	metav1 "k8s.io/apimachinery/pkg/apis/meta/v1"
+)
+
+type EvictedPodsMonitoring struct {
+	logger                     logger.Logger
+	controller                 *Controller
+	evictedPodsCleanupInterval *time.Duration
+	podNamesToDelete           []string
+	stopChan                   chan struct{}
+}
+
+func NewEvictedPodsMonitoring(ctx context.Context,
+	parentLogger logger.Logger,
+	controller *Controller,
+	evictedPodsCleanupInterval *time.Duration) *EvictedPodsMonitoring {
+
+	loggerInstance := parentLogger.GetChild("cron_job_monitoring")
+
+	newEvictedPodsMonitoring := &EvictedPodsMonitoring{
+		logger:                     loggerInstance,
+		controller:                 controller,
+		evictedPodsCleanupInterval: evictedPodsCleanupInterval,
+	}
+
+	parentLogger.DebugWithCtx(ctx, "Successfully created evicted pods monitoring instance",
+		"evictedPodsCleanupInterval", evictedPodsCleanupInterval)
+
+	return newEvictedPodsMonitoring
+}
+
+func (epm *EvictedPodsMonitoring) start(ctx context.Context) {
+
+	// create stop channel
+	epm.stopChan = make(chan struct{}, 1)
+
+	// start a go routine that will periodically clean up stale resources
+	go epm.cleanupEvictedPods(ctx)
+}
+
+func (epm *EvictedPodsMonitoring) stop(ctx context.Context) {
+	epm.logger.DebugWithCtx(ctx, "Stopping evicted pods monitoring")
+
+	if epm.stopChan != nil {
+		epm.stopChan <- struct{}{}
+	}
+}
+
+func (epm *EvictedPodsMonitoring) cleanupEvictedPods(ctx context.Context) {
+	epm.logger.DebugWithCtx(ctx, "Starting evicted pods cleanup")
+
+	// run forever
+	for {
+		select {
+		case <-epm.stopChan:
+			epm.logger.DebugWithCtx(ctx, "Stopping evicted pods cleanup")
+
+			// remove all pods that were marked for deletion
+			epm.podNamesToDelete = nil
+			return
+
+		case <-time.After(*epm.evictedPodsCleanupInterval):
+			epm.logger.DebugWithCtx(ctx, "Cleaning up evicted pods")
+
+			// get all failed function pods
+			stalePodsFieldSelector := common.CompileStalePodsFieldSelector()
+			pods, err := epm.controller.kubeClientSet.CoreV1().Pods("").List(ctx, metav1.ListOptions{
```

Comment:
nuclio controller gets the namespace it should manage, use that and dont rely on its service account to filter out which namespace to track